### ISSUE TITLE
Wrong indent is ignored in YAML files

### ISSUE THREAD

**Issue by skycaptain** [editorconfig-checker/editorconfig-checker#172]
I have the following .editorconfig file:

```ini
root = true

[*]
indent_size = 2
insert_final_newline = true
```

And the following test file (note four spaces for indentation):

```yaml
stages:
    - audit
    - build
    - test
    - docs
```

However editorconfig 2.3.5 does not fail with an error on the wrong indentation (4 instead of 2). It fails however when the newline is missing as expected:

```shell
$ editorconfig-checker -config .editorconfig -verbose
Exclude Regexp: yarn\.lock$|package-lock\.json|composer\.lock$|\.snap$|\.otf$|\.woff$|\.woff2$|\.eot$|\.ttf$|\.gif$|\.png$|\.jpg$|\.jpeg$|\.webp$|\.avif$|\.mp4$|\.wmv$|\.svg$|\.ico$|\.bak$|\.bin$|\.pdf$|\.zip$|\.gz$|\.tar$|\.7z$|\.bz2$|\.log$|\.patch$|\.css\.map$|\.js\.map$|min\.css$|min\.js$
Add .editorconfig to be checked
Add test.yml to be checked
Validate .editorconfig
Validate test.yml
Final newline error found in test.yml
test.yml:
        Wrong line endings or new final newline

1 errors found
2 files checked
```

After fixing newline, but still with wrong indent:

```shell
$ editorconfig-checker -config .editorconfig -verbose
Exclude Regexp: yarn\.lock$|package-lock\.json|composer\.lock$|\.snap$|\.otf$|\.woff$|\.woff2$|\.eot$|\.ttf$|\.gif$|\.png$|\.jpg$|\.jpeg$|\.webp$|\.avif$|\.mp4$|\.wmv$|\.svg$|\.ico$|\.bak$|\.bin$|\.pdf$|\.zip$|\.gz$|\.tar$|\.7z$|\.bz2$|\.log$|\.patch$|\.css\.map$|\.js\.map$|min\.css$|min\.js$
Add .editorconfig to be checked
Add test.yml to be checked
Validate .editorconfig
Validate test.yml
2 files checked
```

**Comment by mstruebing** [comment]
This is not a bug in editorconfig-checker.
This tool only verifies that it uses general formatting rules and has no sense of context at all.
When you would change the spaces to 3 you would see an error, as it is not intended with 2 spaces (multiple amounts of time).
So, if you really need context sensitive formatting/linting you need to go to some more tool to be specialized in your filetype/language.

Sorry.

**Comment by skycaptain** [comment]
Ah, of course makes sense. Thank you.
